pymc/step_methods/arraystep.py

```diff
@@ -50,7 +50,7 @@ class Competence(IntEnum):
 class BlockedStep(ABC):
 
     generates_stats = False
-    stats_dtypes: List[Dict[str, np.dtype]] = []
+    stats_dtypes: List[Dict[str, type]] = []
```

Comment:
FYI @aloctavodia here I made it less strict, because also all the other step methods had things like `np.float64` in the dict but those weren't classified as dtypes either.